src/main/java/ch/heigvd/pro/b04/questions/QuestionController.java

```diff
@@ -100,6 +96,45 @@ public QuestionController(QuestionRepository repository,
         .collect(Collectors.toList());
   }
 
+  /**
+   * gets a {@link ServerQuestion} by his id.
+   *
+   * @param token token of the moderator or participant
+   * @param idModerator id of moderator owning the poll
+   * @param idPoll part of the {@link ServerPollIdentifier}
+   * @param idQuestion part of the {@link ServerQuestionIdentifier}
+   * @return {@link ServerQuestion} found
+   * @throws ResourceNotFoundException if one of the parameters is broken
+   * @throws WrongCredentialsException if there a credentials violation
+   */
+  @GetMapping(value = "/mod/{idModerator}/poll/{idPoll}/question/{idQuestion}")
+  @Transactional
+  public ServerQuestion byId(
+      @RequestParam(name = "token") String token,
+      @PathVariable(name = "idModerator") int idModerator,
+      @PathVariable(name = "idPoll") int idPoll,
+      @PathVariable(name = "idQuestion") int idQuestion)
+      throws ResourceNotFoundException, WrongCredentialsException {
+    ServerQuestionIdentifier idToFind = ServerQuestionIdentifier.builder()
+        .idServerQuestion(idQuestion)
+        .idxPoll(pollRepository.findById(ServerPollIdentifier.builder().idPoll(idPoll).idxModerator(
+            moderatorRepository.findById(idModerator).orElseThrow(ResourceNotFoundException::new))
+            .build())
+            .orElseThrow(ResourceNotFoundException::new))
```

Comment:
Strange indentation, but the code looks correct to me 😄

Reply:
It validates checkstyle :)